### PR TITLE
Developer Guide: Update the Person Role Implementation

### PR DESCRIPTION
**Pull request by purivirakarin**
Update the `Design Consideration` for Person Role Feature

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -250,9 +252,27 @@ The `Person` class is extended by three other classes, each with their own addit
...
-* **Flexible Data Handling**: Fields not applicable to all roles are made optional within the `Person` class to enable a scalable and adaptable system architecture.
-
-* **Role-Specific UI Elements**: The decision to dynamically adjust the UI based on the person's role enhances the overall user experience by providing context-sensitive information.
+**Aspect: How to distinguish the person role:**
+
+* **Alternative 1 (current choice)**: Creation of subclasses `Supplier`, `Employee`, and `Client` that inherit from the `Person` class.
+   * **Pros**:
+       * Specialization: Allows for clear role-specific methods and properties, ensuring that objects have only the attributes and behaviors pertinent to their role.
+       * Extensibility: Easier to add new roles by creating additional subclasses.
+       * Polymorphism: Enables the use of a single interface to represent any person type, which simplifies code that interacts with these objects.
```
Comment (shaunlxw) [inline docs/DeveloperGuide.md]: Can we use ordered list for this to maintain consistency with the format in the `delete` implementation section? 😄 
image

**Review by shaunlxw — CHANGES_REQUESTED** [review]
Mainly LGTM! Just some consistency in formatting, either use ordered list here, or change to unordered list in the delete section.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (purivirakarin) [reply docs/DeveloperGuide.md]: Thanks!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -194,20 +194,27 @@ The sequence diagram below illustrates the creation and execution of a `CsvExpor
...
   * Cons:
-        1. Reduces user flexibility in determining where the CSV file should be saved.
+      1. Reduces user flexibility in determining where the CSV file should be saved.
 * **Alternative 2:** Allow users to specify a file path, defaulting to a pre-defined location if not specified.
   * Pros:
-        1. Provides flexibility for users to save the CSV file wherever they prefer.
+      1. Provides flexibility for users to save the CSV file wherever they prefer.
   * Cons:
-        1. Additional error handling is required to manage invalid or inaccessible file paths.
+      1. Additional error handling is required to manage invalid or inaccessible file paths.
```
Comment (shaunlxw) [inline docs/DeveloperGuide.md:201]: Thanks for fixing the formatting! Actually if its only one thing, we can omit the list I guess..

**Review by shaunlxw — APPROVED** [review]
LGTM! Can consider removing the list completely for export feature.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (purivirakarin) [reply docs/DeveloperGuide.md:201]: Good points. Updated, thanks!

**Pull request merged**